spec/factory_logger_spec.rb

```diff
@@ -94,4 +94,25 @@
                                             ]
     end
   end
+
+  describe ".descendants and .ancestors" do
+    let!(:top) { create :factory_log, name: "top", parent_id: nil }
+    let!(:child1) { create :factory_log, name: "child1", parent_id: top.id }
+    let!(:child2) { create :factory_log, name: "child2", parent_id: top.id }
+    let!(:child3) { create :factory_log, name: "child3", parent_id: top.id }
+    let!(:grand_son1_1) { create :factory_log, name: "grand_son1_1", parent_id: child1.id }
+    let!(:grand_son1_2) { create :factory_log, name: "grand_son1_2", parent_id: child1.id }
+    let!(:grand_son2_1) { create :factory_log, name: "grand_son2_1", parent_id: child2.id }
+
+    it "have descendants" do
+      expect(top.descendants).to eq [
+                                      [child1, [ [grand_son1_1], [grand_son1_2] ] ],
+                                      [child2, [ [grand_son2_1] ] ],
+                                      [child3]
+                                    ]
```

Comment:
Style/IndentArray: Indent the right bracket the same as the start of the line where the left bracket is.